### ISSUE TITLE
Service worker/Workbox doesn't appear to get added when using bundle from widget template on static page

### ISSUE THREAD

**Issue by jcburnette** [preactjs/preact-cli#820]
Hi guys,

Quick question about service workers/workbox. I'm trying to use a widget template and I need a service worker to look for updates. I've implemented a workbox configuration using workbox-webpack-plugin like the following:

In preact.config.js I have:
`config.plugins.push(    
    new WorkboxPlugin.InjectManifest({
      swSrc: 'src/service-worker.js',
      swDest: './service-worker.js',
      include: [/\.js$/]
    })
);`

My service-worker.js file looks like this:

`workbox.setConfig({ debug: true });
workbox.core.skipWaiting();
workbox.core.clientsClaim();

workbox.precaching.precacheAndRoute(self.__precacheManifest || []);`

FInally, in my index.js file, I have:

`if ('serviceWorker' in navigator) {
  const wb = new Workbox('service-worker.js');

  wb.addEventListener('installed', event => {
    if (event.isUpdate) {
      if (confirm('New content is available!. Click OK to refresh')) {
        window.location.reload();
      }
    }
  });
  wb.register();
}`

This works fine when I run preact serve or preact watch, but it doesn't appear to do anything if I build the project and then add it to an html page. I feel like I'm missing something here but I don't know what.

Any information is appreciated.

**Comment by prateekbh** [comment]
what version of preact-cli are you on?

**Comment by robertknight** [comment]
Hello @jcburnette, I've moved this issue to the preact-cli repository because it sounds as though your difficulty is not related to the main Preact library itself.

If you are able to provide us with a runnable project (for example as a GitHub repo) that shows your problem, it may be easier for someone to help.

**Comment by jcburnette** [comment]
Hi guys,

Apologies for the late response. After a bunch of trial and error, I've discovered that the cause of this stems from something in the workbox-window module. I tried to make it work with this module but was not successful. However, upon falling back onto a more traditional way of registering the worker I am up and running again.

For what it's worth, I'm using preact-cli 2.2.1 and preact 8.4.2. I'm not sure why it doesn't seem to register it, but not relying on this module has solved my issue.

Thanks!
